test/perf/perf_test.go

```diff
@@ -0,0 +1,56 @@
+package perf
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	. "github.com/codeready-toolchain/toolchain-e2e/wait"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestPerformances(t *testing.T) {
+	// given
+	ctx, awaitility := WaitForDeployments(t, &toolchainv1alpha1.UserSignupList{})
+	// defer ctx.Cleanup()
+
+	// host metrics should become available at this point
+	metricsService, err := awaitility.Host().WaitForMetricsService("host-operator-metrics")
+	require.NoError(t, err, "failed while waiting for the 'host-operator-metrics' service")
+
+	count := 10
+	t.Run(fmt.Sprintf("%d users", count), func(t *testing.T) {
+		// given
+		users := CreateMultipleSignups(t, ctx, awaitility, count)
+		for _, user := range users {
+			awaitility.Host().WaitForMasterUserRecord(user.Spec.Username, UntilMasterUserRecordHasCondition(Provisioned()))
+		}
+
+		// when deleting the host-operator pod
+		err := awaitility.Host().DeletePods(client.MatchingLabels{"name": "host-operator"})
```

Comment:
Shouldn't we wait for the pod to be actually deleted?
Then wait for the deployment to be green again (for this one not sure because maybe waiting for the metrics service is enough)?